### DISCUSSION TITLE
SIGSEGV php-fpm error

### DISCUSSION THREAD

**Discussion by goldmerc** [libvips/php-vips#190] [Q&A]
php-vips: 2.1.1
libvips42: 8.9.1-2
Ubuntu 20.04

I'm updating to php-vips v2. I'm using Laravel Homestead which has Ubuntu 20.04 installed. If I use apt-get to install libvips42 I appear to get version 8.9.1-2.

Image processing works sporadically. When it doesn't work, I get 502 errors. Looking in the php-fpm log, I see entries like this...

```
[24-Feb-2023 11:48:16] NOTICE: [pool www] child 33136 started
[24-Feb-2023 11:48:20] WARNING: [pool www] child 33136 exited on signal 11 (SIGSEGV - core dumped) after 3.470735 seconds from start
```

After researching online, it seems to me that this may be some kind of memory leak issue which is beyond my understanding! But looking here https://github.com/libvips/php-vips/issues/139 it seems that "libvips 8.13 ought to fix this"

So, first of all, am I correct in my diagnosis? or is the problem something else?

Assuming I am correct above, If I want to install a more up to date version of libvips42 than I am getting from apt-get, is my only option to build from source? I've found the details on the libvips install docs page but it's a bit beyond me! I especially don't understand which dependencies I would need and how to make sure they are installed? I'm only planning to optimise jpeg and webp images for my website. Ideally, it would be great to use mozjpeg. Is there an example anywhere? Thanks

**Comment by jcupitt** [comment]
Hi @goldmerc,

Oooof I always forget this annoying issue. Yes, you'll need a recent libvips with php-fpm, sorry.

You have a few options:

1. Build libvips from source. It's not a hard package to build, but you will need to be comfortable with compilers and linking.
2. Deploy in Docker. You can deploy in a container and use any version of libvips you like. Ubuntu 22.10 comes with libvips 8.13, for example.
3. Update your OS.

**Comment by goldmerc** [comment]
Well, I have no experience building from source but I thought i'd give it a go, following this https://github.com/libvips/libvips#cheatsheet

I'm trying the latest libvips 8.14.1. When I run 'meson setup build', I get this error...

`libvips/meson.build:45:4: ERROR: Dependency "gobject-introspection-1.0" not found, tried pkgconfig`

Also the only optional dependency that the installer can find is libpng. Which dependencies would you recommend I install to be able to handle jpegs and webp? 

If you don't have time to point me in the right direction, no worries. I'm sure you have better things to do! :-) I'll keep chipping away at it

**Reply by jcupitt** [reply]
The notes say -- I would use the meson option `-Dintrospection=false` to turn off introspection support.

For JPEGs you need `libjpeg-dev` and `libexif-dev`. I would use colour management too, so `liblcms2-dev`.

For WEBP you need `libwebp-dev`.

**Reply by goldmerc** [reply]
Sorry John - I'm getting tired and didn't even realise there were instructions above the cheatsheet! I installed libgirepository1.0-dev and introspection is working just fine. I'm making some progress. This is what I'm running...

```
sudo pip3 install meson
sudo apt-get install -y build-essential pkg-config libglib2.0-dev libexpat1-dev libgirepository1.0-dev ninja-build
sudo apt-get install -y libjpeg-dev libexif-dev liblcms2-dev libwebp-dev
wget https://github.com/libvips/libvips/archive/refs/tags/v8.14.1.tar.gz
tar -xvf v8.14.1.tar.gz
cd libvips-8.14.1
meson setup build
cd build
meson compile
meson test
sudo meson install
```

This runs through but when I call vips -v I get this error...

`vips: error while loading shared libraries: libvips.so.42: cannot open shared object file: No such file or directory`

However, if I run tools/vips -v from the build folder, I get...

```
vips-8.14.1
memory: high-water mark 0 bytes
```

Any ideas? Is it something to do with the --prefix flag? I didn't understand how to use it, so I just omitted it!

**Reply by goldmerc** [reply]
hang on! I think running sudo ldconfig did the job :-)

**Comment by goldmerc** [comment]
@jcupitt - just to say thanks. I got it working on both my local and live setups. 👍🏻
